src/python/pants/backend/python/util_rules/dists.py

```diff
@@ -154,6 +155,7 @@ class DistBuildResult:
 wheel_config_settings = {wheel_config_settings_str}
 sdist_config_settings = {sdist_config_settings_str}
 
+os.makedirs(dist_dir, exist_ok=True)
```

Comment:
This is a fresh sandbox each time, so the `exist_ok` should not be necessary in practice. But it's good to be defensive, in the unlikely event that a directory called `dist` does exist in the input snapshot for some reason.